### PR TITLE
[AscendNPU-IR][Developer] Support slice for print and other Ops, complement test cases

### PR DESCRIPTION
**Pull request by shakespearjia**
- support sliced npuir_print for AIV core and buffer print for AIC core
- check and give test cases  for gather/interleave/transpose/pad/flip/bitcast/dot/concat

### CODE REVIEW

File: testing/npuir/test_slice_IR_cmp.py [hunk 1]
```diff
@@ -0,0 +1,391 @@
...
+    N = 512
+    func = vec_bitcast(M, N, 128, 256)
+    kernel = tilelang.engine.lower(func)
+    # print(kernel)
+
+    curr_name = "bitcast" + ".mlir"
+    # Export to .mlir file
+    output_file = './output/' + curr_name
+    with open(output_file, 'w') as f:
+        f.write(kernel)
```
Comment (CeleNewYear) [inline testing/npuir/test_slice_IR_cmp.py]: No Need. We do not use pytest to verify the generated MLIR; this is an unstable CI solution.

File: testing/npuir/test_slice_IR_cmp.py [hunk 2]
```diff
@@ -0,0 +1,391 @@
...
+    N = 64
+    func = vec_concat(M, N, 16, 32)
+    kernel = tilelang.engine.lower(func)
+    # print(kernel)
+
+    curr_name = 'vec_concat' + ".mlir"
+    # Export to .mlir file
+    output_file = './output/' + curr_name
+    with open(output_file, 'w') as f:
+        f.write(kernel)
```
Comment (CeleNewYear) [inline testing/npuir/test_slice_IR_cmp.py]: No Need

File: testing/npuir/test_slice_IR_cmp.py [hunk 3]
```diff
@@ -0,0 +1,391 @@
...
+    N = 512
+    func = vec_flip(M, N, 128, 256)
+    kernel = tilelang.engine.lower(func)
+    # print(kernel)
+
+    curr_name = 'vec_flip' + ".mlir"
+    # Export to .mlir file
+    output_file = ir_dir + curr_name
+    with open(output_file, 'w') as f:
+        f.write(kernel)
```
Comment (CeleNewYear) [inline testing/npuir/test_slice_IR_cmp.py]: No Need

File: testing/npuir/test_slice_IR_cmp.py [hunk 4]
```diff
@@ -0,0 +1,391 @@
...
+    # Export to .mlir file
+    output_file = './output/' + curr_name
+    with open(output_file, 'w') as f:
+        f.write(kernel)
+    
+    ref_file = ir_dir + curr_name
+    # filecmp.cmp returns True if files are identical, False otherwise
+    are_identical = filecmp.cmp(output_file, ref_file , shallow=False)
+    # assertion for pytest
+    assert are_identical, f"'{output_file}' and '{ref_file}' are not identical"
```
Comment (CeleNewYear) [inline testing/npuir/test_slice_IR_cmp.py]: No need, This is an incorrect verification; it does not directly verify accuracy.

File: testing/npuir/test_slice_IR_cmp.py [hunk 5]
```diff
@@ -0,0 +1,391 @@
...
+    print('test sliced vinterleave success')
```
Comment (CeleNewYear) [inline testing/npuir/test_slice_IR_cmp.py]: No need, This is an incorrect verification; it does not directly verify accuracy.

File: testing/npuir/test_slice_IR_cmp.py [hunk 6]
```diff
@@ -0,0 +1,391 @@
...
+    output_file = './output/' + curr_name
+    with open(output_file, 'w') as f:
+        f.write(kernel)
+    
+    ref_file = ir_dir + curr_name
+    # filecmp.cmp returns True if files are identical, False otherwise
+    are_identical = filecmp.cmp(output_file, ref_file , shallow=False)
+    # assertion for pytest
+    assert are_identical, f"'{output_file}' and '{ref_file}' are not identical"
+    print('test sliced vpad success')
```
Comment (CeleNewYear) [inline testing/npuir/test_slice_IR_cmp.py]: No need, This is an incorrect verification; it does not directly verify accuracy.

File: testing/npuir/test_slice_IR_cmp.py [hunk 7]
```diff
@@ -0,0 +1,391 @@
...
+    output_file = './output/' + curr_name
+    with open(output_file, 'w') as f:
+        f.write(kernel)
+    
+    ref_file = ir_dir + curr_name
+    # filecmp.cmp returns True if files are identical, False otherwise
+    are_identical = filecmp.cmp(output_file, ref_file , shallow=False)
+    # assertion for pytest
+    assert are_identical, f"'{output_file}' and '{ref_file}' are not identical"
+    print('test sliced vtranspose success')
```
Comment (CeleNewYear) [inline testing/npuir/test_slice_IR_cmp.py]: No need, This is an incorrect verification; it does not directly verify accuracy.

File: testing/npuir/test_slice_IR_cmp.py [hunk 8]
```diff
@@ -0,0 +1,391 @@
...
+
+
+if __name__ == "__main__":
+    dir_path = Path('./output')
+    if not dir_path.exists():
+        dir_path.mkdir(parents=True)
+        print("create output dir in cur dir")
```
Comment (CeleNewYear) [inline testing/npuir/test_slice_IR_cmp.py]: Delete these, please modify the test file.

File: testing/npuir/test_slice_IR_cmp.py [hunk 9]
```diff
@@ -0,0 +1,391 @@
...
+
+    test_vec_bitcast()
+    test_vec_concat()
+    test_vec_flip()
+    test_vec_gather()
+    test_vec_interleave()
+    test_vec_pad()
+    test_vec_transpose()
```
Comment (CeleNewYear) [inline testing/npuir/test_slice_IR_cmp.py]: Now, different kernel need different name. Please change the kernels' name

File: testing/npuir/test_slice_print.py [hunk 10]
```diff
@@ -0,0 +1,136 @@
...
+    return vecAdd2D
+
+
+def test_vec_add_2d():
+
+    M, N = 256, 256
+
+    A = torch.randn(size=[M, N], dtype=eval("torch." + dtype)).npu()
+    B = torch.randn(size=[M, N], dtype=eval("torch." + dtype)).npu()
+    C = torch.zeros(size=[M, N], dtype=eval("torch." + dtype)).npu()
```
Comment (CeleNewYear) [inline testing/npuir/test_slice_print.py]: Do not use randn, we need to test print, so we need to make sure the printed content is correct.

**Review by CeleNewYear — CHANGES_REQUESTED** [review]

**Review by CeleNewYear — APPROVED** [review]
LGTM

File: testing/npuir/test_slice_concat.py [hunk 11]
```diff
@@ -0,0 +1,61 @@
...
+
+import torch
+import torch_npu
+
+import tilelang
+import tilelang.language as T
+
+tilelang.cache.clear_cache()
+
+torch.npu.set_device(6)
```
Comment (CeleNewYear) [inline testing/npuir/test_slice_concat.py]: change to 0, cause some user do not have enough npu

File: testing/npuir/test_slice_interleave.py [hunk 12]
```diff
@@ -0,0 +1,72 @@
...
+
+import torch
+import torch_npu
+
+import tilelang
+import tilelang.language as T
+
+tilelang.cache.clear_cache()
+
+torch.npu.set_device(6)
```
Comment (CeleNewYear) [inline testing/npuir/test_slice_interleave.py]: change to zero

File: testing/npuir/test_slice_pad.py [hunk 13]
```diff
@@ -0,0 +1,59 @@
...
+
+import torch
+import torch_npu
+
+import tilelang
+import tilelang.language as T
+
+tilelang.cache.clear_cache()
+
+torch.npu.set_device(6)
```
Comment (CeleNewYear) [inline testing/npuir/test_slice_pad.py]: change to zero

File: testing/npuir/test_slice_transpose.py [hunk 14]
```diff
@@ -0,0 +1,56 @@
...
+
+import torch
+import torch_npu
+
+import tilelang
+import tilelang.language as T
+
+tilelang.cache.clear_cache()
+
+torch.npu.set_device(6)
```
Comment (CeleNewYear) [inline testing/npuir/test_slice_transpose.py]: change to zero
